test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -2349,4 +2358,88 @@ protected ClusterState getClusterState() {
         return client(internalCluster().getClusterManagerName()).admin().cluster().prepareState().get().getState();
     }
 
+    /**
+     * Refreshes all indices in the cluster when forceRefresh set to true and
+     * checks if replica shards caught up with primary shard when Segment Replication is enabled.
+     */
+    protected void waitForReplication(boolean forceRefresh, String... indices) {
```

Comment:
The reason I added an extra boolean parameter is because some tests do a flush(), flushAndRefresh() and after that they want to check if replica caught up. In that case it would be easier to just call waitForReplication(false) instead of doing refresh again every time. 

I know doing refresh() everytime doesn't do any harm but I want the waitForReplication() to handle both cases. 

I thought it would would be good to have option of either doing a refresh or skip refresh. This way this method can handle the case of when test doesn't need a refresh again but need to check if replica is caught up with primary

Reply:
I don't have strong opinion having this boolean value other than for reason I said above. but yeah you are right if the test doesn't call refresh/flush manually before then waitForReplication() will fail.

I am also leaning towards what you said, just to keep things simple I think it's worth to do refresh() always rather than relying on test to do it.

Reply:
> some tests do a flush(), flushAndRefresh() and after that they want to check if replica caught up.

Those tests could be changed to do a `flush(); waitForReplication();` and then there won't be a duplicate refresh.